src/main/java/net/mcreator/ui/component/JSingleEntrySelector.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.component;
+
+import com.formdev.flatlaf.FlatClientProperties;
+import net.mcreator.generator.mapping.MappableElement;
+import net.mcreator.minecraft.DataListEntry;
+import net.mcreator.minecraft.MCItem;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.component.util.ComponentUtils;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.init.BlockItemIcons;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.laf.themes.Theme;
+import net.mcreator.util.StringUtils;
+import net.mcreator.util.image.IconUtils;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public abstract class JSingleEntrySelector<T> extends JPanel {

```

Comment:
It may be smart to add support for IValidable here as some may use it in e.g. plugins.

Also, this new component looks like it could be used for other things as well because it is not optimal to have different UI systems that offer more or less the same thing.

Some examples:

* Advancement reward function selector
* GUI bound to selector for block ME
* GUI bound to selector for item ME
* GUI bound to selector for living entity ME

----

It may make sense to move this to own PR and implement it for some existing use cases